newrelic/api/import_hook.py

```diff
@@ -173,25 +168,16 @@ def find_module(self, fullname, path=None):
         self._skip[fullname] = True
 
         try:
-            # For Python 3 we need to use find_loader() from the new
-            # importlib module.
-
-            if find_loader:
-                loader = find_loader(fullname, path)
-
-                if loader:
-                    return _ImportHookChainedLoader(loader)
-
-            else:
-                __import__(fullname)
-
-                # If we get this far then the module we are
-                # interested in does actually exist and so return
-                # our loader to trigger import hooks and then return
-                # the module.
-
-                return _ImportHookLoader()
-
+            try:
+                from importlib.util import find_spec
+                return _ImportHookChainedLoader(find_spec(fullname).loader)
```

Comment:
True. I've added test coverage for this now, and handled the AttributeError case.

Reply:
The try/except cases here may be catching a bit too much: I'm looking at the documentation for [find_spec](https://docs.python.org/3/library/importlib.html#importlib.util.find_spec) and in 3.7+ it says it can raise a `ModuleNotFoundError` (subclass of `ImportError`), which would cause the `__import__(fullname)` code to run.

Looking at the implementation, I think the intent was to only run the "else" case when `from importlib.util import find_spec` raises an import error? If so, we should probably adjust the logic to use the `else` case of the try to run `return _ImportHookChainedLoader(find_spec(fullname).loader)`.

```python
try:
  from importlib.util import find_spec
except ImportError:
  __import__(fullname)
  return _ImportHookLoader()
else:
  spec = find_spec(fullname)
  if spec:
    return _ImportHookChainedLoader(spec.loader)
```

Thoughts on this approach? Should we protecting against the `ValueError` case? You've probably looked into more detail with `find_spec` than I have at this point, and I'm not sure what code would cause a `ValueError` to be raised. 😄